### PR TITLE
Don't show global event listeners in the admin UI

### LINKED ISSUE #34602: Rework global event listener for metrics
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

Leftovers from #33046

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

* Don't show global event listener in the UI
* Make sure the test runs only on Quarkus, as requirements are missing in Undertow

### Actual behavior

* Listener shows in the UI
* Tests would fail on Undertow

### How to Reproduce?

Look at the UI of a realm for event listener

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Closes #34602

### CODE REVIEW

**Comment by ahus1** [comment]
This is my first contribution to the React code of Keycloak. It would be great if someone from the @keycloak/ui could have an extra careful look at this.

Thanks!

cc: @ssilvert

File: js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx [hunk 1]
```diff
@@ -15,8 +26,27 @@ export const EventListenersForm = ({
...
+
+  const { adminClient } = useAdminClient();
+
+  useFetch(
+    async () => {
+      return fetchAdminUI<EventListenerRepresentation[]>(
+        adminClient,
+        "ui-ext/available-event-listeners",
+      );
+    },
```
Comment (edewit) [inline js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx:40]:
```suggestion
    () =>
      fetchAdminUI<EventListenerRepresentation[]>(
        adminClient,
        "ui-ext/available-event-listeners",
      ),
```

File: js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx [hunk 2]
```diff
@@ -15,8 +26,27 @@ export const EventListenersForm = ({
...
+    (data) => {
+      setEventListeners(data);
+    },
```
Comment (edewit) [inline js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx:43]:
```suggestion
      setEventListeners,
```

File: js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx [hunk 3]
```diff
@@ -34,7 +64,10 @@ export const EventListenersForm = ({
           collapsedText: t("showRemaining"),
         }}
         variant={SelectVariant.typeaheadMulti}
-        options={Object.keys(eventListeners!)}
+        options={eventListeners!.map((value) => ({
+          key: value.id,
+          value: value.id,
+        }))}
```
Comment (edewit) [inline js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx:70]:
```suggestion
        options={eventListeners!.map((value) => value.id)}
```

**Review by edewit — CHANGES_REQUESTED** [review]
Some small line savers, otherwise :+1:

File: js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx (hunk 3 above)
Comment (jonkoops) [reply js/apps/admin-ui/src/realm-settings/event-config/EventListenersForm.tsx:70]: You can probably also omit the `!`

**Comment by ahus1** [comment]
@edewit - thank you for the review! I've apply your suggested changes. Ready for another round of review.

**Review by edewit — APPROVED** [review]

**Review by mhajas — APPROVED** [review]
Thank you @ahus1 and @edewit!

**Pull request merged**